lib/Mojolicious/Guides/Tutorial.pod

```diff
@@ -0,0 +1,938 @@
+
+  # Access request information
+  get '/agent' => sub ($c) {
+    my $host = $c->req->url->to_abs->host;
+    my $ua   = $c->req->headers->user_agent;
+    $c->render(text => "Request by $ua reached $host.");
+  };
+
+  # Echo the request body and send custom header with response
+  post '/echo' => sub ($c) {
+    $c->res->headers->header('X-Bender' => 'Bite my shiny metal ass!');
+    $c->render(data => $c->req->body);
+  };
+
+  app->start;
+
+You can test the more advanced examples right from the command line with L<Mojolicious::Command::get>.
+
+  $ ./myapp.pl get -v -M POST -c 'test' /echo
+
+=head2 JSON
+
+JSON is the most commonly used data-interchange format for web services. L<Mojolicious> loves JSON and comes with the
+possibly fastest pure-Perl implementation L<Mojo::JSON> built right in, which is accessible through
+L<Mojo::Message/"json"> as well as the reserved stash value C<json>.
+
+  use Mojolicious::Lite -signatures;
+
+  # Modify the received JSON document and return it
+  put '/reverse' => sub ($c) {
+    my $hash = $c->req->json;
+    $hash->{message} = reverse $hash->{message};
+    $c->render(json => $hash);
+  };
+
+  app->start;
+
+You can send JSON documents from the command line with L<Mojolicious::Command::get>.
+
+  $ ./myapp.pl get -M PUT -c '{"message":"Hello Mojo!"}' /reverse
+
+=head2 Built-in C<exception> and C<not_found> pages
+
+During development you will encounter these pages whenever you make a mistake, they are gorgeous and contain a lot of
+valuable information that will aid you in debugging your application.
+
+  use Mojolicious::Lite -signatures;
+
+  # Not found (404)
+  get '/missing' => sub ($c) {
+    $c->render(template => 'does_not_exist');
+  };
+
+  # Exception (500)
+  get '/dies' => sub { die 'Intentional error' };
+
+  app->start;
+
+You can even use CSS selectors with L<Mojolicious::Command::get> to extract only the information you're actually
+interested in.
+
+  $ ./myapp.pl get /dies '#error'
+
+And don't worry about revealing too much information on these pages, they are only available during development, and
+will be replaced automatically with pages that don't reveal any sensitive information in a production environment.
+
+=head2 Route names
+
+All routes can have a name associated with them, this allows automatic template detection and backreferencing with
+L<Mojolicious::Controller/"url_for">, on which many methods and helpers like
+L<Mojolicious::Plugin::TagHelpers/"link_to"> rely.
+
+  use Mojolicious::Lite -signatures;
+
+  # Render the template "index.html.ep"
+  get '/' => sub ($c) {
+    $c->render;
+  } => 'index';
+
+  # Render the template "hello.html.ep"
+  get '/hello';
+
+  app->start;
+  __DATA__
+
+  @@ index.html.ep
+  <%= link_to Hello  => 'hello' %>.
+  <%= link_to Reload => 'index' %>.
+
+  @@ hello.html.ep
+  Hello World!
+
+Nameless routes get an automatically generated one assigned that is simply equal to the route itself without non-word
```

Comment:
It's not clear to me here or anywhere that I recall...  How does the use of `under` affect auto-generated route names?

```
my $auth = $r->under('/:page);
$auth->get('/:name');
```

What is the auto-generate route name in this case?  Is it `pagename` or `name`?